sound/soc/sof/intel/hda-loader.c

```diff
@@ -551,13 +569,10 @@ int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
 	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_FW_GEN_MSG);
 	msg.primary |= SOF_IPC4_GLB_LOAD_LIBRARY_LIB_ID(fw_lib->id);
 
-	ret = cl_trigger(sdev, hext_stream, SNDRV_PCM_TRIGGER_START);
-	if (ret < 0) {
-		dev_err(sdev->dev, "%s: DMA trigger start failed\n", __func__);
-		goto cleanup;
-	}
-
-	ret = sof_ipc_tx_message_no_reply(sdev->ipc, &msg, 0);
```

Comment:
@lyakh is it not enough to simply call cl_trigger() after sof_ipc_tx_message_no_reply()? But even so I feel like it is not correct. This same sequence works with the ACE firmware when loading external libraries.

Reply:
@ranj063 isn't after `sof_ipc_tx_message_no_reply()` too late already - because it's there where the driver hangs until it times out? Can we double-check with the ACE firmware if it still works now with the current driver version? Then maybe we could check the sources there...